Makefile

```diff
@@ -16,6 +16,7 @@ BUILD_CONTAINER_NAME=calico/cni_build_container
 BUILD_CONTAINER_MARKER=cni_build_container.created
 DEPLOY_CONTAINER_NAME=calico/cni
 DEPLOY_CONTAINER_MARKER=cni_deploy_container.created
+TEMP_DIR:=$(shell mktemp -d)
```

Comment:
I'm slightly on the fence about mentioning this, but you could consider doing the binary inside the deploy container. Then  the build stays more containerized and there's work to do to clean things up or avoid leaks.